qiskit/__init__.py

```diff
@@ -111,15 +115,16 @@ def __bool__(self):
     def __getattr__(self, attr):
         if not self.aer:
             try:
-                from qiskit.providers import aer
+                with warnings.catch_warnings():
+                    warnings.simplefilter("ignore")
+                    from qiskit.providers import aer
```

Comment:
Here (and above), it's probably best to limit this warning filter all the way down to `ignore::DeprecationWarning:qiskit.providers.aer` (or whatever the exact format is), just in case we accidentally mask some real warnings in CI.

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/8657/commits/d36d5dd3ffa06e0a290fd8e069e5f3c1fb743b7d and https://github.com/Qiskit/qiskit-terra/pull/8657/commits/e3512cd7d65be77c1c806e64159d6850fc5de42c